### ISSUE TITLE
Can't fetch small images, API MISUSE, SwiftUI

### ISSUE THREAD

**Issue by zzzkk** [onevcat/Kingfisher#1950]
### Check List

Thanks for considering to open an issue. Before you submit your issue, please confirm these boxes are checked.

- [x] I have read the [wiki page](https://github.com/onevcat/Kingfisher/wiki) and [cheat sheet](https://github.com/onevcat/Kingfisher/wiki/Cheat-Sheet), but there is no information I need.
- [x] I have searched in [existing issues](https://github.com/onevcat/Kingfisher/issues?utf8=✓&q=is%3Aissue), but did not find a same one.
- [x] I want to report a problem instead of asking a question. It'd better to use [kingfisher tag in Stack Overflow](http://stackoverflow.com/questions/tagged/kingfisher) to ask a question.

#### What
Some images fetch always fails, my observation is that it applies only to smaller images like
https://www.google.com/images/branding/googlelogo/1x/googlelogo_color_272x92dp.png
for this https://raw.githubusercontent.com/onevcat/Kingfisher/master/images/logo.png it works fine 🤔  

#### Reproduce
```
KFImage.url(url)
    .onSuccess { r in
        print(r) // never called
    }
    .placeholder { p in
        ProgressView(p) // shows up to 100%
    }
```

#### Other Comment
console output:
```
API MISUSE: NSURLSession delegate KFSessionDelegate: <KFSessionDelegate: 0x28252ac40> (0x28252ac40)
API MISUSE: dataTask:willCacheResponse:completionHandler: completion handler not called
```
- completion for `sessionTask.onTaskDone.delegate(on: self) { (self, done) in` is never called
- using debuging tool I can see successful http response with image

**Comment by onevcat** [comment]
Ummm. It is weird. Kingfisher never implements the `willCacheResponse` delegate method. I will see if I can reproduce it and try to understand what is going on.

**Comment by onevcat** [comment]
Not sure I am doing the right thing or not, but when running in the demo project, it seems everything goes well:

<img width="976" alt="截屏2022-06-08 15 08 35" src="https://user-images.githubusercontent.com/1019875/172543898-9365d5a1-dd47-4dc2-93e9-47cafb57bb84.png">

@zzzkk Is it a 100% reproducible issue? Can you also try on some other simulator/device/network condition and share more information about it? (Or are you trying to extend `KFSessionDelegate` (or `SessionDelegate` in Swift) and implement some `willCacheResponse` method in your project?)

**Comment by onevcat** [comment]
Maybe a reproducible sample repo would be a great help for us to track this issue better!

**Comment by zzzkk** [comment]
right, issue reproduces only in my repo, no matter which device/network/etc
I'm not extending session delegate or modifing other networking related logic

but

maybe some other external library I use is messing it up?
I'll try to turn them off one by one and check if it fixes Kingfisher behaviour 🤔

**Comment by onevcat** [comment]
> maybe some other external library I use is messing it up?

Maybe. But actually, that's beyond what I can imagine... Kingfisher is creating its own `URLSession` when downloading an image and the session delegate is also defined inside the project. So not sure what can cause the "API MISUSE". Please let me know if you can find anything. Thanks!

**Comment by onevcat** [comment]
Thanks for the feedback. Great to hear you found that!

That is a very common mistake that one forgets calling the handler. I guess once DBDebugToolKit fixes it, everything should go back working.
